### PR TITLE
st7789: add scrolling functions to match st7735

### PR DESCRIPTION
**Pull request by deadprogram**
This PR add scrolling functions to the st7789 to the those in the st7735 so tinyterm can work.

### CODE REVIEW

File: st7789/st7789.go [hunk 1]
```diff
@@ -435,6 +435,27 @@ func (d *Device) IsBGR(bgr bool) {
...
+		false)
+}
+
+// SetScroll sets the vertical scroll address of the display.
+func (d *Device) SetScroll(line int16) {
+	d.Command(VSCRSADD)
+	d.Tx([]uint8{uint8(line >> 8), uint8(line)}, false)
+}
+
+// SpotScroll returns the display to its normal state
```
Comment (conejoninja) [inline st7789/st7789.go]: typo Spot -> Stop

File: st7789/st7789.go [hunk 2]
```diff
@@ -435,6 +435,27 @@ func (d *Device) IsBGR(bgr bool) {
 	d.isBGR = bgr
 }
 
+// SetScrollWindow sets an area to scroll with fixed top and bottom parts of the display
```
Comment (conejoninja) [inline st7789/st7789.go]: ScrollWindow -> ScrollArea

**Review by conejoninja — CHANGES_REQUESTED** [review]

File: st7789/st7789.go (hunk 1 above)
Comment (deadprogram) [reply st7789/st7789.go]: Fixed.

File: st7789/st7789.go (hunk 2 above)
Comment (deadprogram) [reply st7789/st7789.go]: Fixed.

**Comment by deadprogram** [comment]
OK all changes made, and a couple of other corrections as well.

**Comment by conejoninja** [comment]
Great!. Nice work, thanks. Merging.

**Pull request merged**